pkg/scheduler/framework/plugins/defaultpodtopologyspread/default_pod_topology_spread.go

```diff
@@ -19,70 +19,237 @@ package defaultpodtopologyspread
+	// postFilterStateKey is the key in CycleState to DefaultPodTopologySpread pre-computed data for Scoring.
+	postFilterStateKey = "PostFilter" + Name
+
+	// When zone information is present, give 2/3 of the weighting to zone spreading, 1/3 to node spreading
+	// TODO: Any way to justify this weighting?
+	zoneWeighting float64 = 2.0 / 3.0
+)
 
 // Name returns name of the plugin. It is used in logs, etc.
 func (pl *DefaultPodTopologySpread) Name() string {
 	return Name
 }
 
+// postFilterState computed at PostFilter and used at Score.
+type postFilterState struct {
+	selector labels.Selector
+}
+
+// Clone implements the mandatory Clone interface. We don't really copy the data since
+// there is no need for that.
+func (s *postFilterState) Clone() framework.StateData {
+	return s
+}
+
 // Score invoked at the Score extension point.
 // The "score" returned in this function is the matching number of pods on the `nodeName`,
 // it is normalized later.
 func (pl *DefaultPodTopologySpread) Score(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+	c, err := state.Read(postFilterStateKey)
+	if err != nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("Error reading %q from cycleState: %v", postFilterStateKey, err))
+	}
+
+	s, ok := c.(*postFilterState)
+	if !ok {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("%+v convert to tainttoleration.postFilterState error", c))
+	}
+
 	nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
 	if err != nil {
 		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
 	}
 
-	meta := migration.PriorityMetadata(state)
-	s, err := pl.calculateSpreadPriorityMap(pod, meta, nodeInfo)
-	return s.Score, migration.ErrorToFrameworkStatus(err)
+	count := countMatchingPods(pod.Namespace, s.selector, nodeInfo)
+	return int64(count), nil
 }
 
 // NormalizeScore invoked after scoring all nodes.
+// For this plugin, it calculates the source of each node
+// based on the number of existing matching pods on the node
+// where zone information is included on the nodes, it favors nodes
+// in zones with fewer existing matching pods.
 func (pl *DefaultPodTopologySpread) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
-	meta := migration.PriorityMetadata(state)
-	err := pl.calculateSpreadPriorityReduce(pod, meta, pl.handle.SnapshotSharedLister(), scores)
-	return migration.ErrorToFrameworkStatus(err)
+	countsByZone := make(map[string]int64, 10)
+	maxCountByZone := int64(0)
+	maxCountByNodeName := int64(0)
+
+	for i := range scores {
+		if scores[i].Score > maxCountByNodeName {
+			maxCountByNodeName = scores[i].Score
+		}
+		nodeInfo, err := pl.handle.SnapshotSharedLister().NodeInfos().Get(scores[i].Name)
+		if err != nil {
+			return migration.ErrorToFrameworkStatus(err)
```

Comment:
replace with ```return framework.NewStatus(framework.Error, err.Error())``` so that we cut the dependency on the migration package which should be deprecated once we finish refactoring everything into plugins.